### ISSUE TITLE
Possibility of taking advantage of rust enums?

### ISSUE THREAD

**Issue by rakshith-ravi** [nicoburniske/leptos_query#2]
We currently have boolean signals for is loading and is fetching.

What if we have an enum with all the states of the data?

```rust
pub enum DataStates<T> {
    Loading,
    Fetching {
        old_data: T
    },
    Loaded {
        is_stale: bool,
        data: T
    }
}
```

Horrible example, but you get the point. This way, certain variables can only exist in certain states, preventing mis-use of data when it cannot exist.

Thoughts?

**Comment by nicoburniske** [comment]
I think this is a great idea. 

I can definitely add this to `QueryResult` and add derived boolean signals for people who only want to use one variation.

**Comment by nicoburniske** [comment]
Closed by #3

**Comment by nicoburniske** [comment]
Closed by #3
